k-distribution/tests/regression-new/locations/test.k

```diff
@@ -0,0 +1,13 @@
+module TEST-SYNTAX
+  imports UNSIGNED-INT-SYNTAX
+  syntax Exp [locations]
+  syntax Exp ::= Exp "+" Exp | Int [klabel(IntConst)]
```

Comment:
What is the purpose of the `IntConst` label?

Reply:
I think, instead of an injection, it adds the label `IntConst`.
But I have a different question, why do you need to add `[locations]` to the sort declaration?
I would expect location information to be accessible to every term everywhere.
Even if the backend doesn't preserve it currently, we can say `N/A` and fix it later.

Reply:
Yes, by why do we need that label? Won't an injection work just as well?

Reply:
If we added it to every term everywhere, it would greatly increase the size of the kore representation. We'd also need to unwrap the source information for every sort.

Reply:
I'm pretty sure you're right that an injection would also work for the location information; I didn't think of this when I was writing the test, though. Do you want me to change it back? Both should work so there's nothing wrong with the test as it is currently written.

Reply:
As far as why we don't add it everywhere, the answer is it will greatly complexify the implementation of the semantics which would then need to know how to deal with location information being wrapped around every single term.